swap/src/protocol/bob.rs

```diff
@@ -38,20 +38,166 @@ mod message3;
 pub mod state;
 pub mod swap;
 
-pub type Swarm = libp2p::Swarm<Behaviour>;
+pub struct Swap {
+    pub state: BobState,
+    pub event_loop_handle: bob::EventLoopHandle,
+    pub db: Database,
+    pub bitcoin_wallet: Arc<bitcoin::Wallet>,
+    pub monero_wallet: Arc<monero::Wallet>,
+    pub swap_id: Uuid,
+}
+
+pub struct SwapFactory {
+    swap_id: Uuid,
+    identity: Keypair,
+    peer_id: PeerId,
+    db_path: PathBuf,
+    config: Config,
+
+    alice_connect_address: Multiaddr,
+    alice_connect_peer_id: PeerId,
+
+    pub bitcoin_wallet: Arc<bitcoin::Wallet>,
+    pub monero_wallet: Arc<monero::Wallet>,
+    pub starting_balances: StartingBalances,
+}
+
+impl SwapFactory {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        seed: Seed,
+        db_path: PathBuf,
+        swap_id: Uuid,
+        bitcoin_wallet: Arc<bitcoin::Wallet>,
+        monero_wallet: Arc<monero::Wallet>,
+        config: Config,
+        starting_balances: StartingBalances,
+        alice_connect_address: Multiaddr,
+        alice_connect_peer_id: PeerId,
+    ) -> Self {
+        let identity = network::Seed::new(seed).derive_libp2p_identity();
+        let peer_id = identity.public().into_peer_id();
+
+        Self {
+            swap_id,
+            identity,
+            peer_id,
+            db_path,
+            config,
+            alice_connect_address,
+            alice_connect_peer_id,
+            bitcoin_wallet,
+            monero_wallet,
+            starting_balances,
+        }
+    }
+
+    pub async fn new_swap_as_bob(
+        &self,
```

Comment:
This change is not easily possible in the current version, because the `SwapFactory` is actually reused. The boundaries of the factory are not defined well (it should not be used for assertions in `Test` but `Test` should store the necessary information itself). I'd say we solve this in #150